test/e2e/performanceprofile/functests/2_performance_update/memorymanager.go

```diff
@@ -0,0 +1,821 @@
+				if len(numaInfo) < 2 {
+					Skip(fmt.Sprintf("This test need 2 Numa nodes. The number of numa nodes on node %s < 2", node.Name))
+				}
+			}
+
+			By("Modifying Profile")
+			for _, node := range workerRTNodes {
+				numaCoreSiblings, err = nodes.GetCoreSiblings(&node)
+			}
+			// Get cpu siblings from core 0, 1
+			for reservedCores := 0; reservedCores < 2; reservedCores++ {
+				cpusiblings := nodes.GetAndRemoveCpuSiblingsFromMap(numaCoreSiblings, reservedCores)
+				reserved = append(reserved, cpusiblings...)
+			}
+			reservedCpus := strings.Join(reserved, ",")
+			for key := range numaCoreSiblings {
+				for k := range numaCoreSiblings[key] {
+					cpusiblings := nodes.GetAndRemoveCpuSiblingsFromMap(numaCoreSiblings, k)
+					isolated = append(isolated, cpusiblings...)
+				}
+			}
+			isolatedCpus := strings.Join(isolated, ",")
+			reservedSet := performancev2.CPUSet(reservedCpus)
+			isolatedSet := performancev2.CPUSet(isolatedCpus)
+
+			hpSize1G := performancev2.HugePageSize("1G")
+			hpSize2M := performancev2.HugePageSize("2M")
+
+			requiredHugepages := &performancev2.HugePages{
+				DefaultHugePagesSize: &hpSize1G,
+				Pages: []performancev2.HugePage{
+					{
+						Count: int32(hpCount),
+						Size:  hpSize2M,
+					},
+				},
+			}
+			profile.Spec.HugePages = requiredHugepages
+
+			if !*profile.Spec.RealTimeKernel.Enabled {
+				profile.Spec.RealTimeKernel = &performancev2.RealTimeKernel{
+					Enabled: pointer.BoolPtr(true),
+				}
+			}
+			if profile.Spec.WorkloadHints == nil {
+				profile.Spec.WorkloadHints = &performancev2.WorkloadHints{
+					RealTime: pointer.Bool(true),
+				}
+			}
+			profile.Spec.CPU = &performancev2.CPU{
+				Reserved: &reservedSet,
+				Isolated: &isolatedSet,
+			}
+			profile.Spec.NUMA = &performancev2.NUMA{
+				TopologyPolicy: &policy,
+			}
+			By("Updating Performance profile")
+			profiles.UpdateWithRetry(profile)
+
+			By("Applying changes in performance profile and waiting until mcp will start updating")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+
+			By("Waiting for MCP being updated")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+		})
+
+		It("[test_id:60545] Reject guaranteed pod requesting resources that cannot be satisfied by 2 numa nodes together", func() {
+			var mm1 MMPod
+			mm1.memory = "200Mi"
+			mm1.cpu = "2"
+			mm1.noOfhpgs = "24Mi"
+			mm1.hpgSize = profile.Spec.HugePages.Pages[0].Size
+			guPod := true
+			testPod := mm1.createPodTemplate(profile, guPod, &workerRTNodes[0])
+			By("creating test pod")
+			err = testclient.Client.Create(context.TODO(), testPod)
+			Expect(err).ToNot(HaveOccurred(), "Failed to created test pod")
+			err = pods.WaitForCondition(testPod, corev1.PodConditionType(corev1.PodFailed), corev1.ConditionFalse, 2*time.Minute)
+			err := checkPodEvent(testPod, "TopologyAffinityError")
```

Comment:
IIUC the test pod requires 24 of 2M HP, but on each of the worker nodes the profile configures only 20 of HP count. This means the non of the nodes is suitable for the testpod. why the pod would be even scheduled? what am I missing here?

Reply:
after discussing this offline with Niranjan, this test covers a bug scenario that verifies that a pod requesting a combination of cpus (that can fit under one NUMA) and HP count that fits under 2 NUMAs should be rejected. 
following this, I suggest adding comments/logs to illustrate the scenario that is meant to be tested based on the corresponding bug.

Reply:
Added detailed comments on the behaviour.